processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -137,14 +137,12 @@ public BeanMappingMethod build() {
                 return null;
             }
 
-            if ( !method.getMappingOptions().isRestrictToDefinedMappings() ) {
```

Comment:
> You are also right 😄. I think that we need to distinguish between composite (multiple) source parameters and single source parameters. For single source parameters, we need to ignore it, otherwise apply it.

I think the distinction is between symmetric and asymmetric (keep on using that term, sorry.)  In would not expect `number` to be mapped in any occasion. Since `number` in `chartEntry` would perhaps reflect the number in the charts, while `number` in `studio` would perhaps be a `street number` or something along that lines.

The risk of such confusion is not so high if you are at the same level in an object graph (source level matches target level) because you are most likely talking about the same things. In an asymmetric situation you are cherry-picking mappings. It would be very risky to assume something about the other mappings at that level..

Hope I make sense somehow.. :smile: